molecule_ec2/cookiecutter/{{cookiecutter.molecule_directory}}/{{cookiecutter.scenario_name}}/create.yml

```diff
@@ -73,53 +73,49 @@
         filters:
           name: "{{ item.image_name }}"
       loop: "{{ molecule_yml.platforms }}"
-      when: item.image is not defined
+      when: item.image_id is not defined
       register: ami_info
 
     - name: Create molecule instance(s)
-      ec2:
+      ec2_instance:
         key_name: "{{ keypair_name }}"
-        image: "{{ item.image
-          if item.image is defined
+        image_id: "{{ item.image_id
+          if item.image_id is defined
           else (ami_info.results[index].images | sort(attribute='creation_date', reverse=True))[0].image_id }}"
         instance_type: "{{ item.instance_type }}"
         vpc_subnet_id: "{{ item.vpc_subnet_id }}"
-        group: "{{ security_group_name }}"
-        instance_tags: "{{ item.instance_tags | combine({'instance': item.name})
+        security_group: "{{ security_group_name }}"
+        region: "{{ item.region
+          if item.region is defined
+          else omit }}"
+        tags: "{{ item.instance_tags | combine({'Name': item.name})
           if item.instance_tags is defined
-          else {'instance': item.name} }}"
+          else {'Name': item.name} }}"
+        filters:
+          instance-state-name: running
+          subnet-id: "{{ item.vpc_subnet_id }}"
+          tag:Name: "{{ item.name }}"
         wait: true
-        assign_public_ip: true
-        exact_count: 1
-        count_tag:
-          instance: "{{ item.name }}"
+        wait_timeout: 300
+        network:
+          assign_public_ip: true
       register: server
       loop: "{{ molecule_yml.platforms }}"
       loop_control:
         index_var: index
-      async: 7200
```

Comment:
Shifting from async to wait will significantly slow down creation of many machines, anything above five.

I also feel that shifting from a timeout of 7200 to 300 should be called out separately. I have no idea why the original timeout was two hours, that seems obscene. But shifting from two hours to five minutes is a substantial change.

AWS documentation states store backed instances are provisioned in "usually less than 5 minutes".
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ComponentsAMIs.html#storage-for-the-root-device

People reported two hours required for Windows instances back in 2009
https://serverfault.com/questions/43884/speeding-up-launch-of-amazon-ec2-windows-instances
However my current job has required creating many Windows instances without anywhere near that level of delay

Terraform uses a timeout of ten minutes, this is a well tested time, I've only once had an issue with this timeout.
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance